rego/rego.go

```diff
@@ -1903,6 +1912,10 @@ func (r *Rego) compileQuery(query ast.Body, imports []*ast.Import, m metrics.Met
 		pkg = r.parsedPackage
 	}
 
+	if r.strict {
+		r.compiler.WithStrict(false)
+	}
```

Comment:
This strikes me as a bit redundant. If `r.strict` is false, we'll not change anything; if it's `true`, we'll disable it here.

So can't we just do
```go
r.compiler.WithStrict(false)
```
here? Or maybe QueryCompiler should always have a non-strict compiler, so we'd deal with that in its method, like here? 🤔

Reply:
Yeah I meant is that if `r.compiler.QueryCompiler()` should always be non-strict, it might as well do that on its own. But I'm not sure how to pull that off cleanly, since if `r.compiler` has strict-mode enabled, deriving a `QueryCompiler` from that shouldn't change the strict-mode of `r.compiler`. You might still want to use that one.

Reply:
Hmmm :thought_balloon:, so do you think the current approach is okay? Otherwise, what is the "dirty" way of doing it? Maybe I can spend a bit of time to see if we can make it work.

LMK what do you think? Thanks!